### PR TITLE
TW5.1.23 Filesystem updates

### PR DESCRIPTION
**Pull request by joshuafontany**
Hi @inmysocks,

I have started the deep refactoring needed to bring Bob's filesystem handling parallel to TW5.1.23 (which I wrote ;) ).

These are the base level changes, and I am still in testing/debugging mode. But everything is going good so far. I need to catch a few vulnerabilities where a user can provide a path to do stuff. That type of input needs sanitized, especially for the OokWiki public-facing context.

Thanks for getting to this when you have time!

### CODE REVIEW

File: WebSockets/NodeServerHandlers.js [hunk 1]
```diff
@@ -676,6 +676,7 @@ if($tw.node) {
   */
   $tw.nodeMessageHandlers.savePluginFolder = function(data) {
     $tw.Bob.Shared.sendAck(data);
+    debugger;
```
Comment (joshuafontany) [inline WebSockets/NodeServerHandlers.js:679]: This, for example, needs manual stepping through in VSCode...

File: FileSystem/MultiWikiAdaptor.js [hunk 2]
```diff
@@ -165,59 +153,88 @@ if($tw.node) {
...
+    if(!$tw.Bob.Wikis[prefix] || $tw.Bob.Wikis[prefix].State !== 'loaded') {
+      return $tw.ServerSide.loadWiki(prefix, finish);
     } else {
-      finish();
+      return finish();
     }
     function finish() {
-      if(tiddler && $tw.Bob.Wikis[prefix].wiki.filterTiddlers($tw.Bob.ExcludeFilter).indexOf(tiddler.fields.title) === -1) {
-        self.getTiddlerFileInfo(new $tw.Tiddler(tiddler.fields), prefix,
+      debugger;
```
Comment (joshuafontany) [inline FileSystem/MultiWikiAdaptor.js:173]: Another debugging hook to remove...

**Comment by inmysocks** [comment]
I will try and at least skim through this today, thank you for your help.

**Comment by joshuafontany** [comment]
> I will try and at least skim through this today, thank you for your help.

Ugh... I misfires the callbacks somehow and it has become a mess. I'm exploring Promises for a few key methods like loadWiki. This should definitely be considered an experimental "draft" at this point. I may have to scrap this and refactir from the master again. :[ We'll see what I can figure out today, as atm it is broken.

I noted you use Promises and arrow functions on the server side of things, so I hope that's a good way to go. Mahalo (thanks)!!

**Comment by inmysocks** [comment]
On the server side I am fine with promises or whatever you want to use.
I did notice that you made a lot of changes that just clean up the code, I have been working on some general code clean up so that is a huge help.

**Comment by joshuafontany** [comment]
I had to go fix bugs in the core, and will need to start from scratch here.

**Pull request closed without merging**